Unity-Technologies/com.unity.multiplayer.samples.coop
# fix: connecting while starting game

### Description (*)

This PR makes it so the closing of character selection and starting of a new game is cancelled when a new player joins or when a player quits. Players now have to re-confirm their choice when it happens so as to be properly informed of it.
This PR also cleans up ServerCharSelectState to align with our coding standards.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2358](https://jira.unity3d.com/browse/MTT-2358)
### Manual testing scenarios

1. Start a game and go to character selection with one or more instances of the game
2. On a second instance, join the game just after the last player hit ready
3. See that the closing of character selection is properly cancelled and every player has to confirm their character again
4. Confirm on all of them
5. Just after the last player confirms, quit the session with one of the clients
6. See that the closing of character selection is properly cancelled again
### Questions or comments

### Contribution checklist
